bridges/MastodonBridge.php

```diff
@@ -17,74 +23,142 @@ class MastodonBridge extends FeedExpander {
 		'norep' => array(
 			'name' => 'Without replies',
 			'type' => 'checkbox',
-			'title' => 'Only return initial toots'
+			'title' => 'Only return statuses that are not replies, as determined by relations (not mentions).'
 		),
 		'noboost' => array(
 			'name' => 'Without boosts',
 			'required' => false,
 			'type' => 'checkbox',
-			'title' => 'Hide boosts'
+			'title' => 'Hide boosts. Note that RSS-Bridge will fetch the original status from other federated instances.'
 			)
 		));
 
-	public function getName(){
-		switch($this->queriedContext) {
-		case 'By username':
-			return $this->getInput('canusername');
-		default: return parent::getName();
-		}
-	}
-
-	protected function parseItem($newItem){
-		$item = parent::parseItem($newItem);
-
-		$content = str_get_html($item['content']);
-		$title = str_get_html($item['title']);
-
-		$item['title'] = $content->plaintext;
-
-		if(strlen($item['title']) > 75) {
-			$item['title'] = substr($item['title'], 0, strpos(wordwrap($item['title'], 75), "\n")) . '...';
-		}
+	const AP_HEADER = array(
+			'Accept: application/activity+json'
+		);
 
-		if(strpos($title, 'shared a status by') !== false) {
-			if($this->getInput('noboost')) {
-				return null;
-			}
-
-			preg_match('/shared a status by (\S{0,})/', $title, $matches);
-			$item['title'] = 'Boost ' . $matches[1] . ' ' . $item['title'];
-			$item['author'] = $matches[1];
-		} else {
-			$item['author'] = $this->getInput('canusername');
-		}
-
-		// Check if it's a initial toot or a response
-		if($this->getInput('norep') && preg_match('/^@.+/', trim($content->plaintext))) {
-			return null;
+	public function getName() {
+		switch($this->queriedContext) {
+			case 'By username':
```

Comment:
I don't think this ever worked because there is no 'By username' context, even from when the bridge was created. Check if the username is not null just like you do for `getURI()`